src/libraries/BitmapLib.sol

```diff
@@ -0,0 +1,43 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+pragma solidity 0.8.21;
+
+/// @title  BitmapLib
+library BitmapLib {
+    function setBit(uint128 bitmap, uint128 index, bool isTrue) internal pure returns (uint128) {
+        if (isTrue) {
+            return bitmap | (uint128(1) << index);
+        } else {
+            return bitmap & ~(uint128(1) << index);
+        }
+    }
+
+    function getBit(uint128 bitmap, uint128 index) internal pure returns (bool) {
```

Comment:
I don't like this. `getBit` returns `bool`. It's a little bit fishy. If you read somewhere within the code `getBit` you expect a number to be returned. Instead you see `bool` and start wondering ... How does this work? Why does it return bool? 
My suggestion would be to either rename it to `isOne` or return a number :)